vim_menutrans_helper.py

```diff
@@ -0,0 +1,228 @@
+                        word = unescape_single_quotes(word)
+                    else:
+                        word = unescape_double_quotes(word)
+                    temp_line += word
+                line = temp_line
+
+            word_list = []
+            # Split line string according to blank without backslash or tab 
+            # ahead
+            for match in re.finditer(r"(?:(?:\\.)|[^ \t])+", line):
+                word_list.append(line[match.start(): match.end()])
+
+            if len(word_list) == 0 or word_list[0] == '"':
+                continue
+            index = 0
+            # Menu_item with "disable" or enable followed close behind should 
+            # be ignored
+            if (word_list[0] in menu_commands and len(word_list) > 1 and 
+                    (word_list[1] == "disable" or word_list[1] == "enable")):
+                continue
+
+            if word_list[0] in sil_commands:
+                index += 1
+            # Blocks needs to be translated without being split by '.'
+            if index < len(word_list) and word_list[index] in menu_commands:
+                to_be_translated_blocks = ""
+                index += 1
+                while index < len(word_list):
+                    if (word_list[index] == '<silent>' or 
+                            word_list[index] == '<script>' or 
+                            (word_list[index][0].isdigit() and 
+                            not word_list[index - 1][0].isdigit())):
+                        index += 1
+                    else:
+                        # Menu_item with "ToolBar" has no following words to be 
+                        # translated
+                        if not ("ToolBar." in word_list[index]):
+                            to_be_translated_blocks = word_list[index]
+                        break
+
+                # Remove the redundant digit added at the head and tail of the 
+                # mid_part
+                if (len(to_be_translated_blocks) > 0 and 
+                        to_be_translated_blocks[0] == '5'):
+                    to_be_translated_blocks = to_be_translated_blocks[1:]
+                if (len(to_be_translated_blocks) > 0 and 
+                        to_be_translated_blocks[-1] == '5'):
+                     to_be_translated_blocks = to_be_translated_blocks[:-1]
+
+                # Menu_item with "ToolBar" constructed a tobe_translated_blocks 
+                # with length equals zero
+                if len(to_be_translated_blocks) == 0:
+                    continue
+
+                # Untranslated word string split by "." but with escaped 
+                # literal dot character"\." together
+                to_be_translated_words = []
+                for match in re.finditer(r"(?:\\.|[^.])+", 
+                        to_be_translated_blocks):
+                    word = to_be_translated_blocks[match.start(): match.end()]
+                    if not (len(word) != 0 and word[0] == '-'):
```

Comment:
Nit: you need to check both `word[0]` and `word[-1]` against '-'. Also, add this document reference: `:see ":help menu-separator"`.